File: src/api/assessment_student_time_accommodation.ts
// Narrow live Instructor time-only configuration for an active roster Student.
import type { AssessmentId } from "../../generated/api/AssessmentId";
import type { CourseInstanceId } from "../../generated/api/CourseInstanceId";
import type { AssessmentStudentTimeAccommodation } from "../../generated/api/AssessmentStudentTimeAccommodation";
import type { SaveAssessmentStudentTimeAccommodationInput } from "../../generated/api/SaveAssessmentStudentTimeAccommodationInput";

export interface AssessmentStudentTimeAccommodationClient {
  readonly getAssessmentStudentTimeAccommodation: (
    course: CourseInstanceId,
    assessment: AssessmentId,
    rosterId: string,
  ) => Promise<AssessmentStudentTimeAccommodation>;
  readonly saveAssessmentStudentTimeAccommodation: (
    course: CourseInstanceId,
    assessment: AssessmentId,
    rosterId: string,
    input: SaveAssessmentStudentTimeAccommodationInput,
  ) => Promise<AssessmentStudentTimeAccommodation>;
}

File: crates/learning-data-access/src/assessment_student_time_accommodation.rs
//! Real per-Student Assessment time configuration access boundary.
use crate::{SessionTokenHash, StoreError};
use async_trait::async_trait;
use question_model::{
    AssessmentId, AssessmentStudentTimeAccommodation, CourseInstanceId,
    SaveAssessmentStudentTimeAccommodationInput,
};

/// Direct-Instructor Course authorization is repeated for every read/save.
#[async_trait]
pub trait AssessmentStudentTimeAccommodationStore: Send + Sync {
    /// Loads one active Student by the existing Course roster identifier.
    async fn student_time_configuration(
        &self,
        token: SessionTokenHash,
        course_instance_id: CourseInstanceId,
        assessment_id: AssessmentId,
        roster_id: String,
        save: Option<SaveAssessmentStudentTimeAccommodationInput>,
    ) -> Result<AssessmentStudentTimeAccommodation, StoreError>;
}
